src/Parser.js

```diff
@@ -662,6 +662,35 @@ Parser.prototype.parseStringGroup = function(modeName, optional) {
 };
 
 /**
+ * Parses a regex-delimited group: the largest sequence of tokens
+ * whose concatenated strings match `regex`. Returns the string
+ * formed by the tokens plus some position information.
+ *
+ * @param {RegExp} regex
+ * @param {string} modeName  Used to describe the mode in error messages
+ */
+Parser.prototype.parseRegexGroup = function(regex, modeName) {
+    var outerMode = this.mode;
+    this.mode = "text";
+    var firstToken = this.nextToken;
+    var lastToken = firstToken;
+    var str = "";
+    while (this.nextToken.text !== "EOF"
+           && regex.test(str + this.nextToken.text)) {
+        lastToken = this.nextToken;
+        str += lastToken.text;
+        this.consume();
+    }
+    if (str === "") {
+        throw new ParseError(
+            "Invalid " + modeName + ": '" + firstToken.text + "'",
+            firstToken);
+    }
+    this.mode = outerMode;
```

Comment:
I know it's unlikely anyone is going to try to look at `this` again after you throw the `ParseError`, but seems like it's good hygenie to reset `this.mode` before you do the `str === ""` test (since I don't think you need the mocked mode anymore after the while loop).